File: app/src/main/java/com/orlinskas/notebook/di/ContextModule.java
package com.orlinskas.notebook.di;

import android.content.Context;

import dagger.Module;
import dagger.Provides;

@NotificationScope
@Module
public class ContextModule {

    private Context context;

    public ContextModule(Context context) {
        this.context = context;
    }

    @Provides
    public Context context() {
        return context;
    }

    @ApplicationContext
    @Provides
    public Context applicationContext() {
        return context.getApplicationContext();
    }
}

File: app/src/main/java/com/orlinskas/notebook/di/ContextModule.kt
package com.orlinskas.notebook.di

import android.content.Context

import dagger.Module
import dagger.Provides

@Module
class ContextModule(private val context: Context) {

    @Provides
    fun context(): Context {
        return context
    }

    @ApplicationContext
    @Provides
    fun applicationContext(): Context {
        return context.applicationContext
    }
}

